core/async.lua

```diff
@@ -25,9 +25,11 @@ function async:DoWithDelay(delay, fn, cb)
   }
 
   task.worker = function()
-    local success, err = coroutine.resume(task.thread)
-    if not success then
-      error(err)
+    if (coroutine.status(task.thread) ~= 'dead') then
```

Comment:
It's subtle, but this isn't the bug. The bug is lower down. Undo this, and check the comment below..

Reply:
The bug is after `task.cb()`. Move the `return` line outside of the `if task.cb` check, but inside the dead coroutine check.